spec.html

```diff
@@ -35555,10 +36024,150 @@ <h1>CharacterClass</h1>
+        <emu-alg>
+          1. Return the CharSet that is the result of evaluating |ClassUnion|.
+        </emu-alg>
+        <p>The production <emu-grammar>ClassContents :: ClassIntersection</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Return the CharSet that is the result of evaluating |ClassIntersection|.
+        </emu-alg>
+        <p>The production <emu-grammar>ClassContents :: ClassSubtraction</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Return the CharSet that is the result of evaluating |ClassSubtraction|.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-classunion">
+        <h1>ClassUnion</h1>
+        <p>The production <emu-grammar>ClassUnion :: ClassRange ClassUnion?</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate |ClassRange| to obtain a CharSet _A_.
+          1. If |ClassUnion| is present, then
+            1. Evaluate |ClassUnion| to obtain a CharSet _B_.
+            1. Return the union of CharSets _A_ and _B_.
+          1. Return _A_.
+        </emu-alg>
+        <p>The production <emu-grammar>ClassUnion :: ClassOperand ClassUnion?</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate |ClassOperand| to obtain a CharSet _A_.
+          1. If |ClassUnion| is present, then
+            1. Evaluate |ClassUnion| to obtain a CharSet _B_.
+            1. Return the union of CharSets _A_ and _B_.
+          1. Return _A_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-classintersection">
+        <h1>ClassIntersection</h1>
+        <p>The production <emu-grammar>ClassIntersection :: ClassOperand `&amp;&amp;` [lookahead != `&amp;] ClassOperand</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate the first |ClassOperand| to obtain a CharSet _A_.
+          1. Evaluate the second |ClassOperand| to obtain a CharSet _B_.
+          1. Return the intersection of CharSets _A_ and _B_.
+        </emu-alg>
+        <p>The production <emu-grammar>ClassIntersection :: ClassIntersection `&amp;&amp;` [lookahead != `&amp;] ClassOperand</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate the |ClassIntersection| to obtain a CharSet _A_.
+          1. Evaluate the |ClassOperand| to obtain a CharSet _B_.
+          1. Return the intersection of CharSets _A_ and _B_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-classsubtraction">
+        <h1>ClassSubtraction</h1>
+        <p>The production <emu-grammar>ClassSubtraction :: ClassOperand `--` ClassOperand</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate the first |ClassOperand| to obtain a CharSet _A_.
+          1. Evaluate the second |ClassOperand| to obtain a CharSet _B_.
+          1. Return the subtraction of CharSet _A_ minus CharSet _B_.
```

Comment:
"The subtraction of" reads very strangely to me. In mathematics I normally hear "set difference", but it's probably better to be more explicit:

```suggestion
          1. Return the CharSet consisting of the elements of _A_ which are not also elements of _B_.
```

Ditto the other places this occurs.

Reply:
Done